### PR TITLE
Add code of conduct

### PR DESCRIPTION
**Pull request by whart222**
## Summary/Motivation:
Adding a Code of Conduct for Pyomo

## Changes proposed in this PR:
This is new

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by DLWoodruff** [comment]
"Talk in person" should be "Have a synchronous conversation" (or perhaps something less geeky that means the same thing, but I'm at a loss for what that would be.)

File: CODE_REVIEW_CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+* Offer alternative implementations, but assume the author already considered
+  them. ("What do you think about using a custom validator here?")
+* Seek to understand the author's perspective.
+* Plan to review the entire pull request.  Do not perform a partial review, and then add more comments later.
+* Sign off on the pull request with a :thumbsup: or "Ready to merge" comment.
+
+Style Comments
+--------------
+
+Pyomo currently does not have required style guidelines.
```
Comment (qtothec) [inline CODE_REVIEW_CODE_OF_CONDUCT.md]: Maybe with the exception that docstrings be correctly parseable by ReadTheDocs?

Comment (mrmundt) [reply CODE_REVIEW_CODE_OF_CONDUCT.md]: But there is a "Coding Standards" area in https://pyomo.readthedocs.io/en/stable/contribution_guide.html - which requires 4 space indents and desires PEP8 standards.

**Comment by blnicho** [comment]
Going to close this for now. We realized that these are not the most recent copies of these documents. We will open a new PR with updated versions after the PMC has a chance to make edits.

**Pull request closed without merging**